test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs

```diff
@@ -791,6 +791,28 @@ internal enum FILE_CREATE_FLAGS
         this.AssertNoDiagnostics();
     }
 
+    [Theory]
+    [InlineData("PCSTR")]
+    [InlineData("PCWSTR")]
+    public async Task CommonTemplatesWorkOnOldNetFramework(string templatedTypeName)
+    {
+        this.compilation = await this.CreateCompilationAsync(MyReferenceAssemblies.NetFramework.Net35);
+        this.generator = this.CreateGenerator(new GeneratorOptions { WideCharOnly = false });
+
+        // TryGenerate returns false for PCSTR and PCWSTR
+        var apiName = templatedTypeName switch
+        {
+            "PCSTR" => "CreateFileA",
+            "PCWSTR" => "CreateFileW",
+            _ => templatedTypeName,
+        };
```

Comment:
I'd like a cleaner approach, but I didn't want to muddle around in how Generator.TryGenerate works without asking first.